std/range/package.d

```diff
@@ -6674,6 +6672,16 @@ private:
     ]));
 }
 
+// Issue 17742
+@safe unittest
+{
+    import std.algorithm.iteration : map;
+    auto ror = 5.iota.map!(y => 5.iota.map!(x => x * y).array).array;
+    assert(ror[3][2] == 6);
+    auto result = ror.transposed;
+    assert(result[3][2] == 6);
```

Comment:
This is actually a really bad test case. It's a matrix that is equal to itself transposed.

I'd rather see an array of array literal, and make sure you don't repeat elements so it's less likely you get a false positive:
```D
[
   [1, 2, 3, 4, 5],
   [6, 7, 8, 9, 10],
   ...
]
```

Reply:
Also, your test should be of `result[2][3]`, as the ror is transposed, so the indexes should be reversed.